### ISSUE TITLE
Support ->models() on suggest & Support "map" for specific models

### ISSUE THREAD

**Issue by davisriska** [babenkoivan/elastic-scout-driver-plus#127]
I would love to see these features available in this package. 
First of all the `->models()` support for suggestions. I dont really have ideas how the specific implementation could be handled, but this is how I currently handle getting suggestions and then mapping them into models. To sum it up, I grab the options and create a new SearchResponse and map them into ['hits']['hits'] and then just use the already built functionality.

```php
use ElasticAdapter\Search\SearchResponse;
use ElasticScoutDriverPlus\Decorators\SearchResult;
use ElasticScoutDriverPlus\Factories\LazyModelFactory;

// ...

$suggestions->suggest('suggestions', [
    'prefix'     => $request->query('value'),
    'completion' => ['field' => 'suggestion'],
]);

$searchResponse = new SearchResponse(
    [
        'hits' => [
            'hits' => $suggestions->execute()->raw()['suggest']['suggestions'][0]['options'],
        ],
    ]
);

$result = new SearchResult(
    $searchResponse,
    new LazyModelFactory(
        $searchResponse,
        (new ReflectionProperty($suggestions, 'modelScope'))->getValue($suggestions)
    )
);

$data = $result->models();
```

The "map" functionality for specific models. Something similar that's available with `->load(['author'], Book::class)` where I could specify `->map(fn(Book $book) => ['name' => $book->title], Book::class)`. Again Im not sure about the specific implementation of this. I would assume it would be something similar to `SearchResult::models()`. Either models() would return already a mapped model if the map is set or it would be a new function, something towards `toArray()` that would then take models and map them.

I don't really have experience with contributing to open source, but if you dont have time for this but could see this being a good addition to this package I would be happy to give a try at this with some general guidance for how you would like to see this implemented.

**Comment by babenkoivan** [comment]
Hey @davisriska, it is an interesting idea, but afaik suggesters return different responses depending on the suggestion type. It might be not easy or even impossible to resolve the related model. I'll look into it and check what can be done here.

**Comment by babenkoivan** [comment]
Hey @davisriska, you can get the related models in the recently released [v4.0.0](https://github.com/babenkoivan/elastic-scout-driver-plus/releases/tag/v4.0.0) as follows:

```php
$suggestions = $searchResult->suggestions();
$models = $suggestions->get('title_suggest')->first()->models();
```

**Comment by davisriska** [comment]
Thanks for taking the time to implement this!
